File: python/src/models/employee.py
from sqlalchemy.dialects.mysql import TINYINT
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from database import Base


class Employee(Base):
    __tablename__ = "employee"
    id_employee = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    name = Column(String(30), nullable=False)
    surname = Column(String(30), nullable=False)
    mail = Column(String(30), nullable=False)
    password = Column(String(30), nullable=False)
    is_active = Column(TINYINT(1), nullable=False, default=0)
    id_center = Column(Integer, ForeignKey("transfusion_center.id_center"), nullable=False)
    transfusion_center = relationship("TransfusionCenter", back_populates="employees", ondelete="CASCADE")

File: nodejs/database/models/employee.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const employee = sequelize.define('employee', {
        id_employee: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING(30),
        },
        surname: {
            type: DataTypes.STRING(30),
        },
        mail: {
            type: DataTypes.STRING(30),
        },
        password: {
            type: DataTypes.STRING(30),
        },
        is_active: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        id_center: {
            type: DataTypes.INTEGER
        }
    }, {
        freezeTableName: true,
        timestamps: false,
        underscored: true,
        name: {
            singular: 'employee',
            plural: 'employee'
        }
    });

    employee.associate = models => {
        employee.belongsTo(models.transfusion_center, {
            foreignKey: 'id_center'
        });
    };

    return employee;
};
